up42/stac.py

```diff
@@ -96,3 +102,67 @@ def extend():
 
     pystac.Item.up42 = Up42ExtensionProvider()  # type: ignore
     pystac.Collection.up42 = Up42ExtensionProvider()  # type: ignore
+
+
+class BulkDeletion:
+    session = base.Session()
+    stac_client = base.StacClient()
+
+    def __init__(self):
+        self._collections: set[pystac.Collection] = set()
+        self._items: set[pystac.Item] = set()
+
+    def add(self, *item_ids: str):
+        """
+        Add item IDs to the collection deletion stash.
+        Collections are deleted as a whole: all items in any affected collection must be included.
+
+        Args:
+            *item_ids: Individual item IDs to add for deletion
+        """
+        items = self.stac_client.get_items(*item_ids)
+
+        for item in items:
+            collection = item.get_parent()
+            if isinstance(collection, pystac.Collection):
+                self._items.add(item)
+                self._collections.add(collection)
+
+    def validate(self) -> Optional[IncompleteCollectionDeletionError]:
+        for collection in self._collections:
+            staged_item_ids = {item_in_collection.id for item_in_collection in collection.get_items()}
+            missing_items = staged_item_ids - set(item.id for item in self._items)
+            if missing_items:
+                error_msg = (
+                    f"Collection '{collection.id}' cannot be deleted because the following items were not included "
+                    f"in the deletion request: {list(missing_items)}. "
+                    f"Collections are deleted as a whole, so all {len(staged_item_ids)} items in collection "
+                    f"'{collection.id}' must be explicitly added for deletion. "
+                    f"Please add the missing items to your deletion request."
+                )
+                return IncompleteCollectionDeletionError(error_msg)
+        return None
+
+    def submit(self):
+        """
+        Submit the bulk deletion request. This will effectively delete all items in the collections in the stash.
+
+        Returns:
+            A dictionary mapping collection IDs to their corresponding DELETE responses.
+        """
+        if not self._collections:
+            raise ValueError("No items to delete. Use add() to add item IDs before submitting.")
+
+        if validation_error := self.validate():
+            self._collections.clear()
```

Comment:
why are items are not cleared?

Reply:
since we are now raising the error, we don't need to clear anymore